### ISSUE TITLE
Notifications issue (Telegram and Whatsapp)

### ISSUE THREAD

**Issue by IzK666** [Freeyourgadget/Gadgetbridge#668]
I'm having issues with the notifications received from Telegram and Whatsapp, and the issue is different in each app.

GB version: Latest, 18.4
Android: Nexus 7 (2013) Lineage 14.1 Nightly
Pebble with Pebblebits 2.9.1
I receive 3 messages from whatsapp:
```
1: One
2: Two
3: Three
```

And that's what appears on pebble:
```
1: One
2: One
3: One
```

On Telegram is a bit different. I received the same 3 messages from the same contact:
```
1: One
2: Two
3: Three
```

And that's what I receive:
```
1: One
2: One
   Two
3: One
   Two
   Three
```
There is something to remark: If I receive a message from another contact, in pebble won't repeat the previous messages from the other contact.

I have tested Telegram on my main phone, a Galaxy Nexus with Cyanogen 11 (KK)
This is what I received on pebble:
```
1: One
2:
3:
```
In that case (the blank ones), the title of the notification is "Telegram" and not the contact!

**Comment by ashimokawa** [comment]
@IzK666 
Thanks for reporting. Unfortunately this is a bit hard to debug since no one of the core team uses your mentioned messengers.

Not that it would change anything, but the latest version released on f-droid is 0.18.5 not 0.18.4. 0.19.0 has been tagged and will be in f-droid as soon as they build us.

**Comment by danielegobbetti** [comment]
It is possible the [Messaging Style](https://developer.android.com/guide/topics/ui/notifiers/notifications.html#style) is being used by these apps and we don't properly parse this. This part of the notifications documentation is new to me.

**Comment by danielegobbetti** [comment]
Can you please test the notification-refactor branch? It should fix your problems.

**Comment by IzK666** [comment]
I can test it, but where I can find the apk file?
I've never compiled anything for android....

**Comment by danielegobbetti** [comment]
If you can't build from source have to wait for it to be released.

**Comment by IzK666** [comment]
SOrry for the delay.

I've tested again with v0.19.1 and still not working (same issues)

**Comment by danielegobbetti** [comment]
It is expected, the code is not merged yet.

**Comment by lazarosfs** [comment]
Yes I have seen this too. It's whatsapp way of keeping messages. When you have one unread notification you get a single notification.
When second arrives you get the first one again and another type of notification second which has all unread notifications in EXTRA_LINES intent flag.
